### PR TITLE
feat : Button Groups

### PR DESCRIPTION
**Pull request by ahmed-osman3**

### CODE REVIEW

File: lib/src/components/buttons/button_group.dart [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+  final bool rounded;
+
+  /// Label for [GroupButton]
+  final String? label;
+
+  /// Optional icon for [GroupButton]
+  final IconData? icon;
+
+  /// Function for when [GroupButton] is clicked.
+  final VoidCallback? onPress;
```
Comment (thelukewalton) [inline lib/src/components/buttons/button_group.dart:82]: This should be `onPressed`  to match the button call

File: lib/src/components/buttons/button_group.dart [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+  /// Returns copy of [GroupButton] with fields.
+  GroupButton copyWith({bool? isFinal, bool? isInitial}) {
+    return GroupButton(
+      key: key,
+      isLarge: isLarge,
+      rounded: rounded,
+      label: label,
+      icon: icon,
+      onPress: onPress,
+      // isFinal: isFinal ?? this.isFinal,
```
Comment (thelukewalton) [inline lib/src/components/buttons/button_group.dart:102]: Remove these comments and in the debug properties below

File: lib/src/components/buttons/button_group.dart [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+  @override
+  void initState() {
+    super.initState();
+    selected = false;
+    controller = MaterialStatesController();
+  }
+
+  void onPress() {
+    widget.onPress!();
```
Comment (thelukewalton) [inline lib/src/components/buttons/button_group.dart:133]: This call is not safe, as `!` removes null-safety. To make this call safe we can change it to:
``` widget.onPress?.call();```
as using `?` maintains null-safety.

File: lib/src/components/buttons/button_group.dart [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+  }
+
+  ButtonStyle getStyle(ZetaWidgetBorder borderType, ZetaColors colors) {
+    final ZetaColorSwatch color =
+        selected ? colors.cool : ZetaColorSwatch.fromColor(colors.black);
+
+    return ButtonStyle(
+      shape: MaterialStateProperty.all(
+        RoundedRectangleBorder(
+          borderRadius: borderType.radius,
```
Comment (thelukewalton) [inline lib/src/components/buttons/button_group.dart:217]: Here we should use `_getRadius(borderType)` to fix the inside corners
image

File: lib/src/components/buttons/button_group.dart [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+          return colors.cool.shade20;
+        }
+        if (selected) return colors.black;
+        return colors.white;
+      }),
+      foregroundColor: MaterialStateProperty.resolveWith<Color?>((states) {
+        if (states.contains(MaterialState.disabled)) {
+          return colors.textDisabled;
+        }
+        if (selected) return color.onColor;
```
Comment (thelukewalton) [inline lib/src/components/buttons/button_group.dart:237]: This should be `colors.black.onColor`

File: example/lib/pages/components/button_example.dart [hunk 6]
```diff
@@ -107,10 +107,20 @@ class _ButtonExampleState extends State<ButtonExample> {
                   Column(children: inputButtons(ZetaWidgetBorder.rounded)),
                   Text('Sharp Buttons', style: ZetaTextStyles.displayMedium),
                   Column(children: inputButtons(ZetaWidgetBorder.sharp)),
-                  Text('Floating Action Buttons', style: ZetaTextStyles.displayMedium),
-                  Text('Tap buttons to change current FAB: ', style: ZetaTextStyles.bodyMedium),
-                  Wrap(children: fabs.divide(SizedBox.square(dimension: 10)).toList()),
-                ].divide(const SizedBox.square(dimension: ZetaSpacing.m)).toList(),
+                  Text('Icon Buttons', style: ZetaTextStyles.displayLarge),
```
Comment (thelukewalton) [inline example/lib/pages/components/button_example.dart:110]: wrong text here, should be 'Group Buttons'

File: lib/src/components/buttons/button_group.dart (hunk 5 above)
Comment (ahmed-osman3) [reply lib/src/components/buttons/button_group.dart:241]: I already set the color to be black on line 215, so will this still be necessary?

Comment (thelukewalton) [reply lib/src/components/buttons/button_group.dart:250]: yeah this ensure it will still work if the values of the token variables change

**Review by thelukewalton — APPROVED** [review]

**Pull request merged**